cardano-cli/src/Cardano/CLI/Shelley/Parsers.hs

```diff
@@ -1612,6 +1600,20 @@ pProtocolVersion =
 parseLovelace :: Atto.Parser Typed.Lovelace
 parseLovelace = Typed.Lovelace <$> Atto.decimal
 
+parseAddress :: Atto.Parser (Typed.Address Typed.Shelley)
+parseAddress = do
+  str <- Text.decodeLatin1 <$> Atto.takeWhile1 (\c -> c == '_' || Char.isAlphaNum c)
```

Comment:
For `parseAddress` and `parseStakeAddress`, we should actually allow characters in the range `[33,126]` since those are the [bounds](https://hackage.haskell.org/package/bech32-1.0.2/docs/src/Codec.Binary.Bech32.Internal.html#humanReadableCharMinBound) on the characters allowed in the human-readable prefix of a Bech32-encoded string.